doc/explanation/styling/design.md

```diff
@@ -0,0 +1,107 @@
+# Designs and Theming
+
+:::{versionadded} 1.0.0
+Designs and theming allow controlling the look and feel of individual components or your entire application.
```

Comment:
Maybe connect to *template* too. For me I want to correct a template, so I'm looking for an explanation for that.

So I probably need to be taught that the starting point is a *design*. A design can have different *themes*. And a template can use the design (and theme).

Maybe even visualize how these things are connect via a graph. Or via an app wrapped in a nice template that uses some desing.